www/activities/diary/js/diary.js

```diff
@@ -170,6 +170,33 @@ app.Diary = {
     el.innerText = dateString;
   },
 
+  sendStatistics: async function () {
+    if (app.Settings.get("statistics", "send-data") != true) {
+      return;
+    }
+
+    let entry = await this.getEntryFromDB(); // Get diary entry from DB
+
+    // Populate groups and get overal nutrition
+    if (entry) {
+      let totalNutrition = await app.FoodsMealsRecipes.getTotalNutrition(entry.items, "ignore");
+
+      let entryDetails = await Promise.all(entry.items.map(async (data) => {
+        return await app.FoodsMealsRecipes.getItem(data.id, data.type, data.portion, data.quantity);
+      }));
+
+      // Send nutrition and diary to target service
+      await app.Utils.timeoutFetch(app.Settings.get("statistics", "send-data-address"), {
```

Comment:
I haven't tested this, bit I think this call would result in an error if the `send-data-address` is not set by the user. Consider checking whether the address is set before making this call.

Reply:
added check for this part. Tested in browser - app does not crash or smth.